dev/devicelab/bin/tasks/dartdocs.dart

```diff
@@ -0,0 +1,52 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:flutter_devicelab/framework/framework.dart';
+import 'package:flutter_devicelab/framework/utils.dart';
+
+Future<Null> main() async {
+  await task(() async {
+    final Stopwatch clock = new Stopwatch()..start();
+    final Process analysis = await startProcess(
+      'flutter',
+      <String>['analyze', '--flutter-repo', '--dartdocs'],
+      workingDirectory: flutterDirectory.path,
+    );
+    int publicMembers = 0;
+    int otherErrors = 0;
+    int otherLines = 0;
+    await for (String entry in analysis.stderr.transform(UTF8.decoder).transform(const LineSplitter())) {
+      print('analyzer stderr: $entry');
+      if (entry.startsWith('[lint] Document all public members')) {
+        publicMembers += 1;
+      } else if (entry.startsWith('[')) {
+        otherErrors += 1;
+      } else {
+        otherLines += 1;
+      }
+    }
+    await for (String entry in analysis.stdout.transform(UTF8.decoder).transform(const LineSplitter())) {
+      print('analyzer stdout: $entry');
+      otherLines += 1;
+    }
+    final int result = await analysis.exitCode;
+    clock.stop();
+    if (publicMembers == 0 && otherErrors == 0 && result != 0)
+      throw new Exception('flutter analyze exited with unexpected error code $result');
```

Comment:
Why not just if `result != 0`?  Or at least return `TaskResult.failure()` down below if `result != 0`?